receiver/otlpreceiver/internal/arrow/arrow.go

```diff
@@ -114,54 +118,57 @@ func (r *Receiver) ArrowStream(serverStream arrowpb.ArrowStreamService_ArrowStre
 	}
 }
 
-func (r *Receiver) processRecords(ctx context.Context, records *arrowpb.BatchArrowRecords) error {
+// processRecords returns an error and a boolean indicating whether
+// the error (true) was from processing the data (i.e., invalid
+// argument) or (false) from the consuming pipeline.  The boolean is
+// not used when success (nil error) is returned.
+func (r *Receiver) processRecords(ctx context.Context, records *arrowpb.BatchArrowRecords) (invalid bool, _ error) {
```

Comment:
suggestion: i know this function signature is a bit wonky because go doesn't have a great way to add an error code on an error, but i'm wondering if we should just do a triple return of `(statusCode, *ErrorCode, error)` which would let us add more error codes easily in the future.